lib/ansible/module_utils/network/skydive/api.py

```diff
@@ -161,3 +280,66 @@ def run(self, ib_spec):
                 result['changed'] = True
 
         return result
+
+
+class skydive_node(skydive_wsclient, skydive_restclient):
+    """ Implements Skydive Node modules """
+
+    def __init__(self, module):
+        self.module = module
+        provider = module.params['provider']
+        super(skydive_node, self).__init__(self.module, **provider)
+
+    def run(self):
+        try:
+            lookup_query = SKYDIVE_GREMLIN_QUERY + "('Name', '{0}')".format(self.module.params['name'])
+            node_exists = self.restclient_object.lookup_nodes(lookup_query)
+
+            if not node_exists and self.module.params['state'] == 'present':
+                self.wsclient_object.connect()
+                self.wsclient_object.start()
+            elif len(node_exists) > 0 and self.module.params['state'] == 'update':
+                self.wsclient_object.connect()
+                self.wsclient_object.start()
+            elif len(node_exists) > 0 and self.module.params['state'] == 'absent':
+                self.module.params['id'] = node_exists[0].__dict__['id']
+                self.wsclient_object.connect()
+                self.wsclient_object.start()
+        except Exception as e:
+            self.module.fail_json(msg=to_text(e))
+        return self.result
+
+
+class skydive_edge(skydive_wsclient, skydive_restclient):
+    """ Implements Skydive Node modules """
+
+    def __init__(self, module):
+        self.module = module
+        provider = module.params['provider']
+
+        super(skydive_edge, self).__init__(self.module, **provider)
+
+    def run(self):
+        try:
+            edge_exists = False
+            edge_query = SKYDIVE_GREMLIN_EDGE_QUERY + "('RelationType', '{0}')".format(self.module.params['relation_type'])
```

Comment:
You can provide "Parent" and "Child" in a Gremlin Has() query. This way, you will not enumerate the full graph....

Reply:
Done